pymc/distributions/continuous.py

```diff
@@ -856,6 +856,15 @@ def logcdf(value, loc, sigma):
             msg="sigma > 0",
         )
 
+    def icdf(value, loc, sigma):
+        res = Normal.icdf((value + 1.0) / 2.0, loc, sigma)
```

Comment:
This is a bit annoying because we don't allow users to create a `HalfNormal` with `loc != 0` but in theory they could call `icdf` or reach this function with a HalfNormal that has a custom non-zero `loc`. 

My question is, will this expression work for non-zero `loc`?

This is also a question for the `HalfCauchy`. 

I wonder if the best solution is to reimplement these RandomVariables directly in PyMC in a way that they don't accept a `loc` argument. This way we don't have to worry about them in the logp/logcdf/icdf/moment functions.

Reply:
The formula I have implemented work well even when `loc != 0` and is consistent with SciPy results:

Testing HalfNormal:
<img width="905" alt="image" src="https://github.com/pymc-devs/pymc/assets/22087841/89851001-5a51-4f2f-930e-b48280c2df03">

Testing HalfCauchy:
<img width="918" alt="image" src="https://github.com/pymc-devs/pymc/assets/22087841/16f5e2c0-9806-4d01-9b7b-8edb76738b45">

Allow me some more time to bring some screenshots calling the implemented functions with `pm.HalfNormal.icdf()` and `pm.HalfCauchy.icdf()`

If you think the best solution is still to reimplement RandomVariables directly in PyMC in a way that they don't accept a loc argument, let me know.

Reply:
We don't need to do it in this PR, but the fact that it isn't being tested in our suite (and not just the new icdf) is already a good argument to drop it

Reply:
Good news that it works though!